nap/cache/base.py

```diff
@@ -28,16 +34,19 @@ def get_timeout_from_header(self, response):
             return int(cache_header_age.group(1))
 
     def get_cache_key(self, model, url):
-
+        """ Cache key format is %(resource_name)s::%(url)s where the URL may be
+          md5 hashed when it exceeds length.  strips white space from cache key as
+           its a control character in memcached.
+        """
         resource_name = model._meta['resource_name']
+        cache_key = "{0}::".format(resource_name)
 
-        key_parts = {
-            'resource_name': resource_name,
-            'url': url,
-        }
-
-        cache_key = "%(resource_name)s::%(url)s" % key_parts
+        if (len(cache_key) + len(url)) > self.cache_max_key_size:
+            cache_key += md5(url.encode('utf-8')).hexdigest()
```

Comment:
Would a `log.debug()` statement be appropriate here to provide some dev visibility into what the original URL was?

Reply:
I think that would be wise ... its a tough debate between info and debug for this one because almost none of our apps log debug statements, and yet it would be nice to see these in our logs.  One argument for info is that most cache keys based on a sane API url would be under 240 characters.  Thoughts?

Reply:
True. Thinking about it more I think info might be more appropriate for the reason you specified. It's useful information to log in a production environment.